File: runner-talk-chat/src/main/kotlin/com/clprunner/runnertalkchat/common/document/MongoBaseDocument.java
package com.clprunner.runnertalkchat.common.document;

import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.ZonedDateTime;

@Getter
public abstract class MongoBaseDocument {

    @Id
    private String id;

    @CreatedDate
    private ZonedDateTime createdAt;

    @LastModifiedDate
    private ZonedDateTime updatedAt;
}

File: runner-talk-chat/src/main/kotlin/com/clprunner/runnertalkchat/common/document/MongoBaseDocument.kt
package com.clprunner.runnertalkchat.common.document

import lombok.Getter
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import java.time.ZonedDateTime

@Getter
abstract class MongoBaseDocument {
    @Id
    private val id: String? = null

    @CreatedDate
    private val createdAt: ZonedDateTime? = null

    @LastModifiedDate
    private val updatedAt: ZonedDateTime? = null
}